pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -190,7 +190,7 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) error {
 	// If this span extended the last span added -- that is, picked up where it
 	// ended and has the same time-bounds -- then we can simply extend that span
 	// and add to its entry counts. Otherwise we need to record it separately.
-	if l := len(s.flushedFiles) - 1; l > 0 && s.flushedFiles[l].Span.EndKey.Equal(span.Key) &&
+	if l := len(s.flushedFiles) - 1; l >= 0 && s.flushedFiles[l].Span.EndKey.Equal(span.Key) &&
```

Comment:
Can this logic be expressed in a way that a future person won’t re-introduce an off-by-one?

- The `-1` isn’t great -- how can a reader infer why that’s there?
- The `>=` isn’t great either. Most loop bounds should have a clean `<`.
- Maybe the several conditions on `s.flushedFiles[l]` should be inside the loop, with a `break` or `continue`.
- Some well-named local vars would increase clarity.

(Bikeshed, can this be something other than `l` which looks like `1`?)

Reply:
Postmortem analysis of the above is kind of pointless unless we can replay the stream of thoughts exactly at the time OBOE was introduced. The choice of the mnemonic to denote the last position _may_ have played some role, but I doubt that was the reason; i.e., `l` usually stands for `length`; I would have used `last`. More likely, the issue is the cognitive dissonance of 0-based and 1-based indexing, both of which are prevalent. E.g., when we talk about nodes in the cluster, we refer to `n1` as the first; there are plenty of other examples.